code/WorkInProgress/construction/tools.dm

```diff
@@ -450,13 +450,49 @@
 	click_delay = 1
 
 	var/selecting = 0
-	var/mode = "floors"
-	var/icons = list("floors" = 'icons/turf/construction_floors.dmi', "walls" = 'icons/turf/construction_walls.dmi', "restore original")
+	var/mode = null
+	var/icons = list("floors", "walls", "restore original")
 	var/marker_class = list("floors" = /obj/plan_marker/floor, "walls" = /obj/plan_marker/wall)
-	var/selected = "floor"
+	/// icon file selected
+	var/selectedicon
+	/// iconstate selected
+	var/selectedtype
+	/// mod to use for generating smoothwalls
+	var/selectedmod
 	// var/pod_turf = 0
 	var/turf_op = 0
 
+	var/list/wallicons = list(\
+	"diner" = 'icons/turf/walls_derelict.dmi',\
+	"martian" = 'icons/turf/walls_martian.dmi',\
+	"shuttle blue" = 'icons/turf/walls_shuttle.dmi',\
+	"shuttle white" = 'icons/turf/walls_shuttle-debris.dmi',\
+	"shuttle dark" = 'icons/turf/walls_shuttle-debris.dmi',\
+	"overgrown" = 'icons/turf/walls_overgrown.dmi',\
+	"meat" = 'icons/turf/walls_meat.dmi',\
+	"ancient" = 'icons/turf/walls_ancient.dmi',\
+	"cave" = 'icons/turf/walls_cave.dmi',\
+	"lead blue" = 'icons/turf/walls_lead.dmi',\
+	"lead gray" = 'icons/turf/walls_lead.dmi',\
+	"lead white" = 'icons/turf/walls_lead.dmi',\
+	"ancient smooth" = 'icons/turf/walls_iomoon.dmi',\
+	)
+	var/list/wallmods = list(\
+	"diner" = "oldr-",\
+	"martian" = "martian-",\
+	"shuttle blue" = "",\
+	"shuttle white" = "shuttle-",\
+	"shuttle dark" = "dshuttle-",\
+	"overgrown" = "root-",\
+	"meat" = "meatier-",\
+	"ancient" = "ancient-",\
+	"cave" = "cave-",\
+	"lead blue" = "leadb-",\
+	"lead gray" = "leadg-",\
+	"lead white" = "leadw-",\
+	"ancient smooth" = "interior-",\
+	)
```

Comment:
pretty sure you don't need to escape these newlines

Reply:
@Skeletonman0